File: Luyencode/CVB2D - Đổi hệ nhị phân sang thập phân/CVB2D.py
# Hàm chuyển đổi từ hệ cơ số 2 sang cơ số 10
def change(input):
    decimal_value = 0
    base_value = 1

    # Duyệt ngược qua từng ký tự trong chuỗi nhị phân
    for i in range(len(input) - 1, -1, -1):
        if input[i] == '1':
            decimal_value += base_value  # Nếu là '1', cộng vào kết quả
        base_value *= 2  # Tăng bậc của cơ số (2^i)

    return decimal_value

# Nhập số lượng test case
N = int(input())

# Duyệt qua từng test case
for _ in range(N):
    input_str = input().strip()  # Đọc chuỗi nhị phân
    print(change(input_str))  # Chuyển đổi nhị phân sang thập phân và in kết quả

File: Luyencode/CVB2D - Đổi hệ nhị phân sang thập phân/CVB2D.cpp
#include <iostream>
#include <string>
using namespace std;

// Hàm chuyển đổi từ hệ cơ số 2 sang cơ số 10
long long change(string& input) {
    long long decimal = 0;
    long long base = 1;

    for (int i = input.length() - 1; i >= 0; --i) {
        if (input[i] == '1') {
            decimal += base;  // Nếu là '1', cộng giá trị vào kết quả
        }
        base *= 2;  // Tăng bậc của cơ số (2^i)
    }
    return decimal;
}

int main() {
    int N;  // Số lượng test case
    cin >> N;
    
    // Duyệt qua từng test case
    for (int i = 0; i < N; ++i) {
        string input;
        cin >> input;
        
        // Chuyển đổi nhị phân sang thập phân và in kết quả
        cout << change(input) << endl;
    }
    
    return 0;
}
